src/csharpExtensionExports.ts

```diff
@@ -20,4 +22,10 @@ export interface CSharpExtensionExports {
     initializationFinished: () => Promise<void>;
     logDirectory: string;
     profferBrokeredServices: (container: GlobalBrokeredServiceContainer) => void;
+    sendRequest: <Params, Response, Error>(
+        type: RequestType<Params, Response, Error>,
+        params: Params,
+        token: vscode.CancellationToken
+    ) => Promise<Response>;
+    serverExecutablePath: () => Promise<string | undefined>;
```

Comment:
> unless you reload / restart the server

That is exactly what I do.

> Is there a way to check if another extension has a specific export function without activating the other extension itself?

I'm unsure, but I doubt it. These export objects are the returns from `activate`.

Reply:
> Was wondering if it would be possible for other extensions to export a well known function / property that tells us where we can go look for additional parts to load.

We could absolutely have extensions stick something in their package.json which lets us know they'd be exporting such a function. In @333fred's case though, I really would hope he doesn't need to write a custom function, as opposed to simply having static metadata that lets us discover his DLL and pull it into the composition. I'm not saying the custom function may not need to exist, but unless @333fred is trying to do something far more complex than I remember, we should be able to keep a simple path.

Reply:
Static metadata would work.

Reply:
Updated to use this approach.